### PR TITLE
chore: upgrade oclif to v3 and swc to latest

### PR DESCRIPTION
**Pull request by thomaswhyyou**
### Description
This PR takes care of a few major/minor dependency upgrades that we've been punting on for some time:
* Upgrades [Oclif](https://oclif.io/) (the CLI framework we use) to the latest major version (v3) release, following this [migration guide](https://github.com/oclif/core/blob/main/guides/V3_MIGRATION.md).
  * There were some breaking changes but not a ton that affected us. A couple required changes [here](https://github.com/knocklabs/knock-cli/pull/290#discussion_r1488528638) and [here](https://github.com/knocklabs/knock-cli/pull/290#discussion_r1488532351).
* Upgrades all of [SWC](https://swc.rs/) deps to the latest versions

### Tasks
[KNO-5233](https://linear.app/knock/issue/KNO-5233/[kaizen]-upgrade-oclif-to-v3)
[KNO-4909](https://linear.app/knock/issue/KNO-4909/[kaizen]-upgrade-swc-versions-used-in-cli)

### CODE REVIEW

File: .tool-versions
Comment (thomaswhyyou) [inline .tool-versions:1]: This now is the minimum required version by Oclif after the upgrade.

File: bin/dev.js
Comment (thomaswhyyou) [inline bin/dev.js:1]: This was one change we needed to make from the oclif v3 migration guide: https://github.com/oclif/core/blob/main/guides/V3_MIGRATION.md#bin-scripts-for-esmcjs-interoperability

To be clear, all the +/- files in `bin/` are copied straight from oclif generated files.

File: test/commands/commit/get.test.ts
Comment (thomaswhyyou) [inline test/commands/commit/get.test.ts:21]: This was another breaking change from the oclif migration guide: https://github.com/oclif/fancy-test?tab=readme-ov-file#v3-breaking-changes

The oclif test helper lib now uses `sinon` under the hood for stubbing, and it's now expected to use the provided stub object instead of calling `sinon.stub()` directly.

**Review by connorlindsey — APPROVED** [review]
I'm not super familiar with the CLI so feel free to get a second review, but looking at the migration guide and changes here, looks good to me!

**Comment by thomaswhyyou** [comment]
> I'm not super familiar with the CLI so feel free to get a second review, but looking at the migration guide and changes here, looks good to me!

Thanks for taking a look @connorlindsey! We have a good amount of tests in the CLI which helps, but I'm going to do a round of manual QA before cutting a new release to be extra safe.

**Pull request merged**
